plugins/out_s3/s3_store.h

```diff
@@ -28,13 +28,15 @@ struct s3_file {
     int failures;                    /* delivery failures */
     size_t size;                     /* file size */
     time_t create_time;              /* creation time */
+    time_t first_log_time;           /* first log time */
     flb_sds_t file_path;             /* file path */
     struct flb_fstore_file *fsf;     /* reference to parent flb_fstore_file */
 };
 
 int s3_store_buffer_put(struct flb_s3 *ctx, struct s3_file *s3_file,
                         const char *tag, int tag_len,
-                        char *data, size_t bytes);
+                        char *data, size_t bytes,
+                        time_t *time);
```

Comment:
your declared function signature should match your defined function signature. in otherwords, you need to make sure the .c and .h file functions look the same.

Reply:
It's `time` here and `timep` in the other file